src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -1100,14 +1100,21 @@ export const selectStakingOpportunitiesDataFullByFilter = createCachedSelector(
   selectStakingDataByFilter,
   selectAssetIdParamFromFilter,
   selectDefaultStakingDataByValidatorId,
+  selectFeatureFlags,
   (
     portfolioValidatorIds,
     validatorsData,
     stakingDataByValidator,
     assetId,
     defaultStakingData,
+    featureFlags,
   ): OpportunitiesDataFull[] => {
-    if (defaultStakingData) {
+    // used to enable/disable staking by feature flag
+    const featureFlagFilter = () => {
+      if (assetId === osmosisAssetId && !featureFlags.OsmosisStaking) return false
+      return true
+    }
```

Comment:
No need for this to be a function.

```suggestion
    const featureFlagFilter = !(assetId === osmosisAssetId && !featureFlags.OsmosisStaking)
```